scipy/special/orthogonal.py

```diff
@@ -525,12 +540,454 @@ def h_roots(n, mu=False):
+    df = lambda x: 1.0 - cos(x)
+    xi = 0.5*pi
+    for i in range(maxit):
+        xi = xi - f(xi)/df(xi)
+    return xi
+
+
+def initial_nodes_a(n, k):
+    """Tricomi initial guesses
+
+    Computes an initial approximation to the square of the `k`-th
+    (positive) root :math:`x_k` of the Hermite polynomial :math:`H_n`
+    of order :math:`n`. The formula is the one from lemma 3.1 in the
+    original paper. The guesses are accurate except in the region
+    near :math:`\sqrt{2n + 1}`.
+
+    Parameters
+    ----------
+    n : int
+        Quadrature order
+    k : ndarray of type int
+        Index of roots to compute
+
+    Returns
+    -------
+    xksq : ndarray
+        Square of the approximate roots
+
+    See Also
+    --------
+    initial_nodes
+    h_roots_asy
+    """
+    tauk = compute_tauk(n, k)
+    sigk = cos(0.5*tauk)**2
+    a = n % 2 - 0.5
+    nu = 4.0*floor(n/2.0) + 2.0*a + 2.0
+    # Initial approximation of Hermite roots (square)
+    xksq = nu*sigk - 1.0/(3.0*nu) * (5.0/(4.0*(1.0-sigk)**2) - 1.0/(1.0-sigk) - 0.25)
+    return xksq
+
+
+def compute_am(m):
+    """Approximation of the roots of the Airy function
+
+    Computes an asymptotic approximation to the root :math:`a_m`
+    of the Airy function :math:`Ai(x)` for large values of :math:`m`.
+    This formula is essentially exact for :math:`m > 10`.
+
+    Parameters
+    ----------
+    m : ndarray of type int
+        Index of the root :math:`a_m`
+
+    Returns
+    -------
+    am : ndarray
+        The `m`-th root :math:`a_m` of the function :math:`Ai(x)`
+
+    See Also
+    --------
+    airy_root
+    h_roots_asy
+    """
+    sm = 3*pi*(4*m-1) / 8.0
+    coeffs = row_stack([1.0, 5.0/48.0, -5.0/36.0, 77125.0/82944.0,
+                        -108056875.0/6967296.0, 162375596875.0/334430208.0])
+    smp = column_stack([ones_like(sm), sm**(-2), sm**(-4), sm**(-6), sm**(-8), sm**(-10)])
+    am = -sm**(2.0/3.0) * dot(smp, coeffs).reshape((-1,))
+    return am
+
+
+def airy_root(m):
+    """Approximation of the roots of the Airy function
+
+    Computes an approximation to the `m`-th root :math:`a_m` of the
+    Airy function :math:`Ai(x)`. For :math:`m < 10` we use a lookup
+    table and an asymptotic expansion otherwise.
+
+    Parameters
+    ----------
+    m : ndarray of type int
+        Index of the root :math:`a_m`
+
+    Returns
+    -------
+    am : ndarray
+        The `m`-th root :math:`a_m` of the function :math:`Ai(x)`
+
+    See Also
+    --------
+    h_roots_asy
+    """
+    airyroots = array([
+             -2.3381074104597670385,
+             -4.0879494441309706166,
+             -5.5205598280955510591,
+             -6.7867080900717589988,
+             -7.9441335871208531231,
+             -9.0226508533409803802,
+             -10.040174341558085931,
+             -11.008524303733262893,
+             -11.936015563236262517,
+             -12.828776752865757200])
+    # Note: This specialized code is much faster than 'ai_zeros'
```

Comment:
Yes, this function is necessary because it is much faster.
You can try what happens if we take the scipy routine.
I looked into the FORTRAN code and it seems to apply the same asymptotic formula but then doing some more stuff
in a second loop.